controller/internal/supervisor/supervisor.go

```diff
@@ -219,24 +219,30 @@ func (s *Config) doCreatePU(contextID string, pu *policy.PUInfo) error {
 func (s *Config) doUpdatePU(contextID string, pu *policy.PUInfo) error {
 
 	s.Lock()
+	defer s.Unlock()
 
-	data, err := s.versionTracker.LockedModify(contextID, revert, 1)
+	data, err := s.versionTracker.Get(contextID)
```

Comment:
Given the global lock .. I think we don't need the LockedModify any more and we should instead try only f we succeed updating the rules do we change the version.